scvi/train/_trainer.py

```diff
@@ -1,22 +1,21 @@
 import sys
 import warnings
-from typing import Any, List, Literal, Optional, Union
+from typing import Literal, Optional, Union
 
 import numpy as np
 import pytorch_lightning as pl
 from pytorch_lightning.loggers import LightningLoggerBase
 
 from scvi import settings
-from scvi._decorators import classproperty
-from scvi.autotune._types import Tunable
+from scvi.autotune._types import Tunable, TunableMixin
 
 from ._callbacks import LoudEarlyStopping
 from ._logger import SimpleLogger
 from ._progress import ProgressBar
 from ._trainingplans import PyroTrainingPlan
 
 
-class Trainer(pl.Trainer):
+class Trainer(pl.Trainer, TunableMixin):
```

Comment:
why is the mixin going last here? This applies to a lot of them.

Reply:
Should it be going first? Wasn't sure if the order mattered

Reply:
It does matter if there were collisions in method names (lookup MRO python), it's probably best to have the mixins before the base class that provides most functionality